mungegithub/mungers/milestone-maintainer.go

```diff
@@ -35,54 +35,179 @@ import (
 	githubapi "github.com/google/go-github/github"
 )
 
+type milestoneState int
+
+type milestoneStateConfig struct {
+	title          string
+	label          string
+	warnOnInterval bool
+	notifySIGs     bool
+}
+
 const (
 	milestoneNotifierName = "MilestoneNotifier"
 
-	milestoneRemoved          = "Milestone Removed"
-	milestoneLabelsIncomplete = "Milestone Labels **Incomplete**"
-	milestoneLabelsComplete   = "Milestone Labels **Complete**"
+	milestoneModeDev    = "dev"
+	milestoneModeSlush  = "slush"
+	milestoneModeFreeze = "freeze"
+
+	milestoneCurrent milestoneState = iota
+	milestoneNeedsLabeling
+	milestoneNeedsApproval
+	milestoneNeedsAttention
+	milestoneNeedsRemoval
+
+	statusMilestoneRemovedLabel = "status/milestone-removed"
+	statusLabelsIncompleteLabel = "status/incomplete-labels"
+	statusNeedsApprovalLabel    = "status/needs-approval"
+	statusNeedsAttentionLabel   = "status/needs-attention"
+	statusApprovedLabel         = "status/approved-for-milestone"
+
+	blockerLabel = "priority/critical-urgent"
+
+	milestoneDetail = `<details>
+<summary>Help</summary>
+<ul>
+ <li>Additional instructions available <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/release/issues.md">here</a></li>
+ <li>The commands available for setting labels are documented <a href="https://github.com/kubernetes/test-infra/blob/master/commands.md">here</a></li>
+</ul>
+</details>
+`
+
+	milestoneMessageTemplate = `
+{{- if .warnUnapproved}}
+**Action required**: This issue must have the {{.approvedLabel}} label applied by a SIG maintainer.{{.unapprovedRemovalWarning}}
+{{end -}}
+{{- if .removeUnapproved}}
+**Important**: This issue was missing the {{.approvedLabel}} label for more than {{.approvalGracePeriod}}.
+{{end -}}
+{{- if .warnMissingStatus}}
+**Action required**: Must specify at most one of {{.statusLabels}}.
+{{end -}}
+{{- if .warnUpdateRequired}}
+**Action Required**: This issue is marked as {{.blockerLabel}}, but has not been updated in {{.lastUpdated}}. Please provide an update.
```

Comment:
Using the number of days since the last update here will cause the comment update interval to be ignored if it is greater than 24 hours because the notification context will change every 24 hours, forcing `notificationIsCurrent` to return false. 

If `.lastUpdated` is changed from the number of days since the last update to the actual last update time then this won't be a problem.